### PR TITLE
dead code cleanup - dce/kerberos and gridproxy/gss

### PR DESCRIPTION
**Pull request by vchlum**
#### Bug/feature Description
* dead code cleanup in preparation of new gss/kerberos support

#### Affected Platform(s)
* All

#### Cause / Analysis / Design
* [forum](http://community.pbspro.org/t/kerberos-support)
* remove staled gridproxy/gss and dce/kerberos code

#### Solution Description
* remove dce/kerberos:
```
deprecated macros:
PBS_CRED_DCE_KRB5
PBS_CREDNAME_KRB5
PBS_CREDNAME_DCE_KRB5
KAUTH
KCRYPT

removed attribute:
src/include/job.h: ji_credrtime

removed functions:
src/cmds/qsub.c: get_cred_from_cache()
src/cmds/qsub.c: fwd_tgt_creds()
src/cmds/qsub.c: get_krb5_ticket()
src/resmom/requests.c: set_kerb_cred()
src/server/process_request.c: fwd_tgt_creds()
src/server/process_request.c: get_kerb_cred()
src/server/req_quejob.c: renew_cred_server()
src/server/req_quejob.c: next_cred_renew()
src/server/req_quejob.c: renew_kerb_cred()
src/server/req_quejob.c: renew_credential()
src/server/req_quejob.c: save_kerb_cred()

removed tool:
src/cmds/pbs_renew.c
doc/man1/pbs_renew.1B

removed file:
src/lib/Libsec/cs.c

```
* remove gridproxy/gss:
```
deprecated macros:
PBS_CRED_GRIDPROXY
PBS_BATCH_GSS_Context
IS_GSS_HANDSHAKE
IS_CLUSTER_KEY
PBS_CREDTYPE_GRIDPROXY

removed attributes or structures:
src/include/batch_request.h: rq_gssdata
src/include/job.h: ji_gsscontext 
src/include/pbs_nodes.h: msr_gsscontext
src/resmom/mom_server.c: svr_context
src/resmom/mom_server.c: pbs_sisterkey

removed functions:
src/cmds/qsub.c: get_grid_proxy()
src/lib/Libifl/int_jcred.c: PBSD_GSS_context()
src/lib/Libifl/int_jcred.c: add_status()
src/lib/Libifl/int_jcred.c: pbs_gss_error()
src/resmom/mom_server.c: setup_gss()
src/resmom/mom_server.c: unseal_server()
src/resmom/requests.c: set_gridproxy()
src/resmom/requests.c: cleanup_cred()
src/server/job_func.c: pbs_freecontext()
src/server/job_func.c: job_freecontext()
src/server/node_manager.c: setup_gss()
src/server/req_quejob.c: unseal_gridproxy()
src/server/req_quejob.c: req_gsscontext()
```

#### Testing logs/output
* [ptl_smoke.txt](https://github.com/PBSPro/pbspro/files/2510115/ptl_smoke.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/include/libpbs.h [hunk 1]
```diff
@@ -262,7 +262,7 @@ struct batch_reply {
 #define PBS_BATCH_JobCred		62
 #define PBS_BATCH_CopyFiles_Cred	63
 #define PBS_BATCH_DelFiles_Cred		64
-#define PBS_BATCH_GSS_Context		65
+#define PBS_BATCH_GSS_Context		65 /* Depricated */
```
Comment (mike0042) [inline src/include/libpbs.h]: Spelling: Deprecated

File: src/include/net_connect.h [hunk 2]
```diff
@@ -101,8 +101,8 @@ typedef unsigned long pbs_net_t;        /* for holding host addresses */
 #define IS_SHUTDOWN		8
 #define IS_IDLE			9
 #define IS_ACKOBIT		10
-#define IS_GSS_HANDSHAKE	11
-#define IS_CLUSTER_KEY		12
+#define IS_GSS_HANDSHAKE	11	/* Depricated */
+#define IS_CLUSTER_KEY		12	/* Depricated */
```
Comment (mike0042) [inline src/include/net_connect.h]: Deprecated

File: src/include/ticket.h [hunk 3]
```diff
@@ -50,28 +50,16 @@ extern "C" {
 
 #define	PBS_CREDVER		1
 #define	PBS_CREDTYPE_NONE	0
-#define	PBS_CREDTYPE_DCE_KRB5	1
-#define	PBS_CREDTYPE_GRIDPROXY	2
+#define	PBS_CREDTYPE_DCE_KRB5	1 /* Depricated */
+#define	PBS_CREDTYPE_GRIDPROXY	2 /* Depricated */
```
Comment (mike0042) [inline src/include/ticket.h]: Deprecated

File: src/include/libpbs.h (hunk 1 above)
Comment (vchlum) [reply src/include/libpbs.h]: Fixed

File: src/include/net_connect.h (hunk 2 above)
Comment (vchlum) [reply src/include/net_connect.h]: Fixed and also fixed spelling of comment on IS_HPCBP_ATTRIBUTES line, which misled me.

File: src/include/ticket.h (hunk 3 above)
Comment (vchlum) [reply src/include/ticket.h]: Fixed

**Comment by mike0042** [comment]
The new changes look good @vchlum. Looks like the Appveyor build needs to be restarted. I usually just re-sign my commit when that happens...
git commit --amend --no-edit -S
git push -f origin kerberos_staled_code

**Comment by subhasisb** [comment]
Appveyor actually failed - here is the last portion of the compilation logs:

dis_read.c
catch_child.c
153>..\..\src\resmom\catch_child.c(1239) : warning C4101: 'vnl_changes' : unreferenced local variable
153>..\..\src\resmom\catch_child.c(1397) : warning C4101: 'extval' : unreferenced local variable
attr_recov.c
Generating Code...
Compiling...
vnparse.c
vnode_storage.c
Generating Code...
Linking...
LINK : fatal error LNK1181: cannot open input file 'c:\projects\win_build\src\lib\Libpbs\Release\Libpbs.lib'
Build log was saved at "file://c:\projects\win_build\src\resmom\Release\BuildLog.htm"
resmom - 1 error(s), 184 warning(s)
Build complete: 13 Projects succeeded, 64 Projects failed, 0 Projects skipped
Command exited with code 1

**Comment by subhasisb** [comment]
HI @vchlum please rebase your branch and squash the commits for merge

**Comment by vchlum** [comment]
Commit has been squashed and rebased. Waiting for appveyor to finish the check.

**Comment by subhasisb** [comment]
Travis failed on Ubuntu due a url timing out - restarted job

**Comment by subhasisb** [comment]
Codacy was hung, merged anyway - codacy had passed earlier anyways

**Pull request merged**